File: Timofeev PE-42/Bizzare Underground C++/Project1/WallBlock.cpp
#include "WallBlock.h"
#include <iostream>

WallBlock::WallBlock() : isHide(true) {
    std::cout << "WallBlock created. Initial state: isHide = " << std::boolalpha << getIsHide() << std::endl;
}

WallBlock::~WallBlock() {
    std::cout << "WallBlock destroyed. Final state: isHide = " << std::boolalpha << getIsHide() << std::endl;
}

void WallBlock::putingWall() {
    std::cout << "WallBlock::putingWall() called. Current isHide: " << std::boolalpha << getIsHide() << std::endl;
    setIsHide(false); // Блок становится осязаемым
    std::cout << "WallBlock::putingWall(): Block extended (made tangible). New state: isHide = " << std::boolalpha << getIsHide() << std::endl;
}

bool WallBlock::getIsHide() const { return isHide; }
void WallBlock::setIsHide(bool hide) {
    isHide = hide;
    std::cout << "WallBlock::setIsHide() called. New state: isHide = " << std::boolalpha << isHide << std::endl;
}

File: Timofeev-PE-42/Bizzare-Underground-Java/WallBlock.java
public class WallBlock extends GameObject {
    private boolean isHide;

    public WallBlock() {
        super(); // Вызываем конструктор базового класса
        this.isHide = true; // По умолчанию блок скрыт
        System.out.println("WallBlock created. Initial state: isHide = " + isHide);
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("WallBlock destroyed. Final state: isHide = " + isHide);
        super.finalize();
    }

    public void putingWall() {
        System.out.println("WallBlock.putingWall() called. Current isHide: " + isHide);
        setIsHide(false); // Блок становится осязаемым
        System.out.println("WallBlock.putingWall(): Block extended (made tangible). New state: isHide = " + isHide);
    }

    // Getter
    public boolean getIsHide() { return isHide; }

    // Setter
    public void setIsHide(boolean hide) {
        this.isHide = hide;
        System.out.println("WallBlock.setIsHide() called. New state: isHide = " + isHide);
    }
}